src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -152,6 +187,27 @@ public ExecutableSequence step() {
     return eSeq;
   }
 
+  /**
+   * Updates eSeq's weight in weightMap according to the Orienteering formula from the GRT paper.
+   *
+   * @param eSeq the recently executed sequence which needs a weight update
+   */
+  private void setWeight(ExecutableSequence eSeq) {
+    SequenceExecDetails seqExecDets = sequenceExecutionCount.get(eSeq.sequence);
```

Comment:
Abstract out much of the body of this into an `orienteeringWeight` routine, and let `setWeight` call that.
This refactoring will ease future generalizations to handle new weighting schemes.